### PR TITLE
Add another reference for twist transformation. Comment correction.

### PR DESCRIPTION
**Pull request by AndyZe**

### CODE REVIEW

File: tf2_eigen/include/tf2_eigen/tf2_eigen.hpp [hunk 1]
```diff
@@ -131,7 +131,7 @@ void doTransform(
 /** \brief Apply a geometry_msgs TransformStamped to a 6-long Eigen vector.
  * Useful for transforming wrenches or twists.
  * Wrench: (force-x, force-y, force-z, torque-x, torque-y, torque-z)
- * Twist: (trans. vel. x, trans. vel. y, trans. vel. z, ang. vel. x, ang. vel. y, ang. vel. z)
+ * Twist: (ang. vel. x, ang. vel. y, ang. vel. z, trans. vel. x, trans. vel. y, trans. vel. z)
```
Comment (AndyZe) [inline tf2_eigen/include/tf2_eigen/tf2_eigen.hpp:134]: If you look at Eq. 3.83 here, this was wrong.

http://hades.mech.northwestern.edu/images/7/7f/MR.pdf

**Comment by AndyZe** [comment]
@ibrahiminfinite @sea-bass

File: tf2_eigen/include/tf2_eigen/tf2_eigen.hpp [hunk 2]
```diff
@@ -148,7 +148,9 @@ void doTransform(
   const geometry_msgs::msg::TransformStamped & transform)
 {
   // References:
-  // https://core.ac.uk/download/pdf/154240607.pdf, https://www.seas.upenn.edu/~meam520/notes02/Forces8.pdf
+  // https://core.ac.uk/download/pdf/154240607.pdf
+  // https://www.seas.upenn.edu/~meam520/notes02/Forces8.pdf
```
Comment (sea-bass) [inline tf2_eigen/include/tf2_eigen/tf2_eigen.hpp]: This URL is now dead, so I think we can get rid of it

Comment (ibrahiminfinite) [reply tf2_eigen/include/tf2_eigen/tf2_eigen.hpp]: Its available on archive. 
But the things in that reference is more confusing. The modern robotics book is better.

[Forces8.pdf](https://github.com/ros2/geometry2/files/12370578/Forces8.pdf)
Here is the pdf if you did not already get it.

**Review by sea-bass — APPROVED** [review]
Following the code and the new reference, I think this comment correction is good.

**Review by ahcorde — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19349/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/13876/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20070/)

**Comment by ahcorde** [comment]
https://github.com/Mergifyio backport humble iron

**Pull request merged**